benchmark/sirun/encoding/index.js

```diff
@@ -0,0 +1,47 @@
+const {
+  ENCODER_VERSION
+} = process.env
+
+const { AgentEncoder } = require(`../../../packages/dd-trace/src/encode/${ENCODER_VERSION}`)
+const id = require('../../../packages/dd-trace/src/id')
+
+const writer = {
+  flush: () => {}
+}
+
+function createSpan (parent) {
+  const spanId = id()
+  return {
+    trace_id: parent ? parent.trace_id : spanId,
+    span_id: spanId,
+    parent_id: parent ? parent.parent_id : id(0),
+    name: 'this is a name',
+    resource: 'this is a resource',
+    error: 0,
+    start: 1415926535897,
+    duration: 100,
+    meta: {
+      a: 'b',
+      hello: 'world',
+      and: 'this is a longer string, just because we want to test some longer strongs, got it? okay'
+    },
+    metrics: {
+      b: 45,
+      something: 98764389,
+      afloaty: 203987465.756754
+    }
+  }
+}
+
+const trace = []
+for (let parent = null, i = 0; i < 30; i++) {
+  const span = createSpan(parent)
+  trace.push(span)
+  parent = span
+}
+
+const encoder = new AgentEncoder(writer)
+
+for (let j = 0; j < 10000; j++) {
+  encoder.encode(trace)
```

Comment:
does the encoder maintain any kind of state between calls to encode?

I am thinking for v0.5 endpoint, does this scenario showcase the benefits we gain from that string table? (does it matter? is that a completely different scenario?)

Reply:
It should show that value. We already see differences in instruction counts between 0.4 and 0.5.